courses/tasks.py

```diff
@@ -0,0 +1,68 @@
+"""
+Tasks for the courses app
+"""
+import logging
+from mitxpro.celery import app
+from mitxpro.utils import now_in_utc
+from courses.models import CourseRun, CourseRunCertificate
+from courses.utils import ensure_course_run_grade, process_course_run_grade_certificate
+from courseware.api import get_edx_grades_with_users
+
+log = logging.getLogger(__name__)
+
+
+@app.task
+def generate_course_certificates():
+    """
+    Task to generate certificates for courses.
+    """
+    now = now_in_utc()
+    course_runs = CourseRun.objects.filter(end_date__lt=now).exclude(
+        id__in=CourseRunCertificate.objects.values_list("course_run__id", flat=True)
+    )
+
+    for run in course_runs:
+        try:
+            edx_grade_user_iter = get_edx_grades_with_users(run)
+        except Exception as exp:  # pylint: disable=broad-except
+            log.exception(
+                "Unable to fetch user grades for run %s from openEdx: %s", run, exp
+            )
+        else:
```

Comment:
Minor: instead of having this very large block of code in an `else` clause, we should just `continue` after logging the exception above. It won't be necessary to run the `log.info` line if an exception occurs